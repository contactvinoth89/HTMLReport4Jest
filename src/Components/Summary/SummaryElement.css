.dot a {
    text-decoration: none;
    font-size: calc(10vmin);
    color: white;
    display: block;
    position: absolute;
    transform: translate(-50%, -50%);
    top: 50%;
    left: 50%;
  }
  
  .dot span{
    text-decoration: none;
    font-size: calc(3vmin);
    color: white;
    display: block;
    position: absolute;
    transform: translate(-50%, -50%);
    top: 80%;
    left: 50%;
  }
  
  .dot a:hover {
    color: #f1f1f1;
  }
  .dot {
    height: 100px;
    width: 100px;
    border-radius: 50%;
    display: inline-block;
    padding: 25px 25px 25px 25px;
    margin: 15px 15px 15px 15px;
    position: relative;
  }
  
  .green{
    background-color: green;
  }
  
  .orange{
    background-color: orange;
  }
  .red{
    background-color: red;
  }
  .blue{
    background-color: blue;
  }
  .gray{
    background-color:gray;
  }