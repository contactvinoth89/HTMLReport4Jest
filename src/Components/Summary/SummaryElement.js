import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './SummaryElement.css';

export default class SummaryElement extends Component {
    render() {
        return (
            <span className={`dot ${this.props.styleClass}`}>
                <a href={`#${this.props.Category}`}>{this.props.data}</a>
                <span>{this.props.Category}</span>
            </span>
        );
    }
}
SummaryElement.propTypes = {
    Category: PropTypes.string.isRequired,
    styleClass: PropTypes.string.isRequired,
    data: PropTypes.number.isRequired,
};
