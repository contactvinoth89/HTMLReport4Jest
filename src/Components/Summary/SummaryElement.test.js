import React from 'react';
import SummaryElement from './SummaryElement';
import { render, waitFor } from '@testing-library/react';
test('should contain chart', async () => {
    const data = {
        labels: ['Passed', 'Pending', 'Failed', 'Todo'],
        datasets: [
            {
                data: [1, 2, 3, 4],
                backgroundColor: ['green', 'orange', 'red', 'gray'],
                hoverBackgroundColor: ['green', 'orange', 'red', 'gray'],
            },
        ],
    };
    const { container } = render(
        <SummaryElement data={data} title="Test title"></SummaryElement>,
    );
    const canvas = await waitFor(() => container.firstChild.firstChild);
    const img = canvas.toDataURL();
    const imageData = img.replace(/^data:image\/\w+;base64,/, '');
    const buf = Buffer.from(imageData, 'base64');
    expect(buf).toMatchSnapshot();
});

test('should contain no data found', () => {
    const data = {
        labels: ['Passed', 'Pending', 'Failed', 'Todo'],
        datasets: [
            {
                data: [0, 0, 0, 0],
                backgroundColor: ['green', 'orange', 'red', 'gray'],
                hoverBackgroundColor: ['green', 'orange', 'red', 'gray'],
            },
        ],
    };
    const { container } = render(
        <SummaryElement data={data} title="Test title"></SummaryElement>,
    );
    expect(container.firstChild).toMatchSnapshot();
});
