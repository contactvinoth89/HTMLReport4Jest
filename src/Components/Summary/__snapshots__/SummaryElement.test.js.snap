// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should contain chart 1`] = `
<div
  class="card"
>
  <canvas
    height="150"
    width="300"
  />
</div>
`;

exports[`should contain no data found 1`] = `
<div
  class="card"
>
  <canvas
    height="150"
    width="300"
  />
</div>
`;
